src/evict/evict_lru.c

```diff
@@ -1893,20 +1912,7 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
         FLD_SET(walk_flags, WT_READ_PREV);
     /* FALLTHROUGH */
     case WT_EVICT_WALK_RAND_NEXT:
-        read_flags = WT_READ_CACHE | WT_READ_NO_EVICT | WT_READ_NO_GEN | WT_READ_NO_WAIT |
-          WT_READ_NOTFOUND_OK | WT_READ_RESTART_OK;
-        if (btree->evict_ref == NULL) {
-            for (;;) {
-                /* Ensure internal pages indexes remain valid */
-                WT_WITH_PAGE_INDEX(session,
-                  ret = __wt_random_descent(session, &btree->evict_ref, read_flags, &session->rnd));
-                if (ret != WT_RESTART)
-                    break;
-                WT_STAT_CONN_INCR(session, cache_eviction_walk_restart);
-                WT_STAT_DATA_INCR(session, cache_eviction_walk_restart);
-            }
-            WT_RET_NOTFOUND_OK(ret);
-        }
+        FIND_RANDOM_PAGE_IF_NONE();
```

Comment:
```   
    /*
     * Get some more eviction candidate pages, starting at the last saved point. Clear the saved
     * point immediately, we assert when discarding pages we're not discarding an eviction point, so
     * this clear must be complete before the page is released.
     */
    ref = btree->evict_ref;
    btree->evict_ref = NULL;
```
We set btree->evict_ref to NULL. Does this mean we always start from random position after this change?

Reply:
No, the changes go above this line.

Reply:
But if we call this function again, it will start from random position because we have set btree->evict_ref to NULL to NULL here, is that correct?

Reply:
`evict_ref` gets assigned back [later in the same function](https://github.com/wiredtiger/wiredtiger/blob/515662c80f79cfbc8f78f56781bf7db0b4d46e1b/src/evict/evict_lru.c#L2182).

Reply:
Thank you. That makes sense.